### PR TITLE
School Mapping Results

### PR DESCRIPTION
**Pull request by mahesh989**
Hello  
 I have completed the project, and things are looking good. Please check complete_match.csv in the results folder for the detailed matched records.

In addition to the initial matching, I also matched names from list_A with old_school_name1 and old_school_name2 in list_B to improve accuracy. I noticed that in some cases, district and root names matched, but the school level didn't, so these additional steps have significantly tightened things up.

I would like  have your say on this. Thanks 
Regards 
Mahesh Tiwari

### CODE REVIEW

File: 2024-05_school_mapping/solutions/solution2/requirements.txt
Comment (sumanashrestha) [inline 2024-05_school_mapping/solutions/solution2/requirements.txt:1]: please check if all these dependencies are actually necessary for this project

**Comment by sumanashrestha** [comment]
Thank you again for your effort, results look good. except for requirements.txt looks good to merge

File: 2024-05_school_mapping/solutions/solution2/requirements.txt
Comment (mahesh989) [reply 2024-05_school_mapping/solutions/solution2/requirements.txt:1]: I have updated the dependencies actually needed for this project.

**Comment by mahesh989** [comment]
> Thank you again for your effort, results look good. except for requirements.txt looks good to merge. The match contains about 11k schools out of 28k schools to match. are there any areas we can focus on to increase matched count?

To answer your question one possibility is to decrease the root_name_score  in analysis_1,2,3.py code. 
# Root name match
    if root_name_score >= 85:
        root_name_score = 2  # Complete match
        match_type = 'complete'
   
   With more that 85 , I am pretty confident with the matching names, as  I manually checked the samples at multiple places. 
   
   With less than 85, I haven't checked but there could be a possibility of a false positive. 
   
   We can do further analysis by excluding those 11k entries and changing  the  score 
   root_name_score >= 80 or 75:
   
Perhaps we can extract more matching names but need to double-check for accuracy. This is my understanding right now.

**Pull request merged**